shell_sort/ShellSort.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+// Use '-std=c++11' or higher flag while compiling
+// g++ -std=c++11 -o ShellSort ShellSort.cpp
+
+using namespace std;
+
+// Worst case time complexity = O(n^2)
+// Best case complexity = O(nlog(n))
+
+vector<int> shellSort(vector<int> data) {
```

Comment:
Shouldn't function name in c++ be separated using underscores rather than using `camelCase`.